spectacles/runner.py

```diff
@@ -267,6 +296,18 @@ async def checkout_temp_branch(self, ref: str) -> str:
         self.is_temp_branch = True
         return name
 
+    async def checkout_ephemeral_branch(
+        self, ref: str, use_personal_branch: bool, update_self_branch: bool = True
+    ) -> None:
+        """Either check out temp or personal branch and hard-reset."""
+        if use_personal_branch:
+            if update_self_branch:
+                self.branch = await self.checkout_personal_branch(ref)
+            else:
+                await self.checkout_personal_branch(ref)
+        else:
+            self.branch = await self.checkout_temp_branch(ref)
+
```

Comment:
My suggestion would be to return the branch and use that to update `self.branch` externally where necessary instead of having this function update `self.branch` on its own and needing the flag and extra conditional logic.